src/query/src/promql/planner.rs

```diff
@@ -1803,6 +1803,12 @@ impl PromPlanner {
                 fn_name: SPECIAL_HISTOGRAM_QUANTILE.to_string(),
             })?
             .clone();
+        // remove le column from tag columns
+        let tag_columns = std::mem::take(&mut self.ctx.tag_columns);
+        self.ctx.tag_columns = tag_columns
+            .into_iter()
+            .filter(|col| col != LE_COLUMN_NAME)
+            .collect();
```

Comment:
Can we use retain?
https://doc.rust-lang.org/stable/std/vec/struct.Vec.html#method.retain